Mathlib/Algebra/Module/Basic.lean

```diff
@@ -332,6 +332,21 @@ def RingHom.toModule [Semiring R] [Semiring S] (f : R →+* S) : Module R S :=
   Module.compHom S f
 #align ring_hom.to_module RingHom.toModule
 
+/-- A homomorphism between semirings R and S can be equivalently described as a R-module S
+  such that S/S/R forms a scalar tower. -/
+def ringHomEquivModuleIsScalarTower [Semiring R] [Semiring S] :
+    (R →+* S) ≃ {inst : Module R S // IsScalarTower R S S} where
+  toFun f := let m := Module.compHom S f; ⟨m, ⟨fun r ↦ mul_assoc (f r)⟩⟩
+  invFun := fun ⟨_, _⟩ ↦
+    { toFun := fun r ↦ r • 1
+      map_zero' := zero_smul R 1
+      map_one' := one_smul R 1
+      map_add' := (add_smul · · 1)
+      map_mul' := fun r r' ↦ by simp_rw [← smul_eq_mul, smul_assoc, one_smul] }
```

Comment:
We already have `smulOneHom`; can you rename it to `MonoidHom.smulOneHom` and extract what you have here to `RingHom.smulOneHom`?

Reply:
Good to know, will do!

Reply:
Mind spinning this out to a helper PR?

Reply:
Is that necessary? Only about 10 lines in total ... (see [this commit](https://github.com/leanprover-community/mathlib4/pull/9046/commits/39a958e6f2dfb597602d713c41279f409401e2ac))

Reply:
Done in #9064